### PR TITLE
Add spell check capability

### PR DESCRIPTION
**Pull request by playpeace**
- Project specific words can be added to a dictionary
- Any typos are treated as build errors

Signed-off-by: Sumit Ahuja <ext-sahuja@luxoft.com>

### CODE REVIEW

File: docs/check_spelling.sh
Comment (tobsan) [inline docs/check_spelling.sh]: Is there a specific reason why the dictionary is a dot-file? I'm thinking for usability reasons it would make sense to just call it `aspell.en.pws` (without the leading dot) or something like that.

File: docs/check_spelling.sh
Comment (tobsan) [inline docs/check_spelling.sh]: If it is possible to use several extensions, there are some index files that are called `.rsti` in the blueprint.

File: docs/check_spelling.sh
Comment (tobsan) [inline docs/check_spelling.sh]: What does this mean? A comment wouldn't hurt!

File: docs/check_spelling.sh
Comment (tobsan) [inline docs/check_spelling.sh]: Is this for autocorrect? A comment wouldn't hurt here either.

File: docs/check_spelling.sh
Comment (tobsan) [inline docs/check_spelling.sh]: Some short comments on why things are run in a certain order would be good here.

**Review by tobsan — CHANGES_REQUESTED** [review]

File: docs/check_spelling.sh
Comment (playpeace) [reply docs/check_spelling.sh]: Done.

File: docs/check_spelling.sh
Comment (playpeace) [reply docs/check_spelling.sh]: Now supports multiple extensions

File: docs/check_spelling.sh
Comment (playpeace) [reply docs/check_spelling.sh]: Done

File: docs/check_spelling.sh
Comment (playpeace) [reply docs/check_spelling.sh]: Yes it is. Updated comments.

File: docs/check_spelling.sh
Comment (playpeace) [reply docs/check_spelling.sh]: Added.

**Comment by playpeace** [comment]
@tobsan @rpannek I can build successfully on my local, but it fails here.
Do you have any idea why ?

File: Jenkinsfile
Comment (tobsan) [inline Jenkinsfile:29]: Even though spell check defaults to on, I like the verbosity here!

File: docs/aspell.en.pws
Comment (tobsan) [inline docs/aspell.en.pws:5]: misspelled word

File: docs/aspell.en.pws
Comment (tobsan) [inline docs/aspell.en.pws:56]: should be cross-compiling

File: docs/aspell.en.pws
Comment (tobsan) [inline docs/aspell.en.pws:61]: Misspelled word?

File: docs/aspell.en.pws
Comment (tobsan) [inline docs/aspell.en.pws:71]: Shouldn't it be space here?

File: docs/aspell.en.pws
Comment (tobsan) [inline docs/aspell.en.pws:94]: should be space?

File: docs/aspell.en.pws
Comment (tobsan) [inline docs/aspell.en.pws:192]: misspelled word

File: docs/aspell.en.pws
Comment (tobsan) [inline docs/aspell.en.pws:193]: Misspelled word

File: docs/aspell.en.pws
Comment (tobsan) [inline docs/aspell.en.pws:194]: Misspelled word

File: docs/aspell.en.pws
Comment (tobsan) [inline docs/aspell.en.pws:217]: One of these are misspelled

File: docs/aspell.en.pws
Comment (tobsan) [inline docs/aspell.en.pws:284]: Misspelled word

File: docs/aspell.en.pws
Comment (tobsan) [inline docs/aspell.en.pws:296]: Misspelled word

**Review by tobsan — APPROVED** [review]
Looks good. I commented on some words that are misspelled. Do we want to fix them in this commit or do that later?

File: docs/aspell.en.pws
Comment (playpeace) [reply docs/aspell.en.pws]: I was planning to do fix them in another commit :)

**Comment by playpeace** [comment]
I was planning to fix the misspelling in another commit :)

**Comment by playpeace** [comment]
@tobsan Could you merge this if it looks okay? I suppose I won't have write access to this.

File: README.md
Comment (jonte) [inline README.md]: I'm guessing that the location of the project specific dictionary is important in order for the `check_spelling.sh` script to find it? Can you mention where to put the dict for project specific checks?

Comment (jonte) [reply README.md]: Also, I believe the location of the symlink is also important (providing that you try finding the project specific dict based on the location of the symlink) - can you mention where the symlink should be created in the project specific case?

File: README.md
Comment (jonte) [inline README.md]: Can we have double dashes for `--usage` ? Also, the comvention is to call such a flag `--help` - but I am OK w/ usage.

If you want a single dash altenrative, I suggest `-u` or `-h`.

File: README.md
Comment (jonte) [inline README.md]: Please add a note on increasing the word count (top of the aspell dict file) when adding new words manually as well. I am not sure what the significance is of the word count, but aspell seems to maintain that number when aspell itself adds words.

File: docs/CMakeLists.txt
Comment (jonte) [inline docs/CMakeLists.txt]: Maybe we can make the `check_spelling.sh` script executable, and add a hashbang at the top of it, then we don't need to invoke "/bin/bash" manually here.

File: docs/check_spelling.sh
Comment (jonte) [inline docs/check_spelling.sh:1]: I would add some copyright here.

File: docs/check_spelling.sh
Comment (jonte) [inline docs/check_spelling.sh]: Space before "It".

Comment (jonte) [reply docs/check_spelling.sh]: and also before "--no-verify"

**Review by jonte — CHANGES_REQUESTED** [review]
Very nice! :D
I commented for some small improvements, but in general - awesome!

**Comment by tobsan** [comment]
@playpeace I'm fine with fixing the spelling in another PR.

File: README.md
Comment (playpeace) [reply README.md]: Guess we should support both short and long options.

File: README.md
Comment (playpeace) [reply README.md]: This is not really needed; The word count is more like a hint.

http://aspell.net/man-html/Format-of-the-Personal-and-Replacement-Dictionaries.html#Format-of-the-Personal-and-Replacement-Dictionaries

Comment (jonte) [reply README.md]: Ah! Good info. Thanks :)

File: docs/CMakeLists.txt
Comment (playpeace) [reply docs/CMakeLists.txt]: Sure, there's already a hashbang, will update the cmake file.

Comment (rpannek) [reply docs/CMakeLists.txt]: Also not all systems have bash installed in `/bin/bash` and I'd advise to use `/usr/bin/env bash` instead, especially in the shebang line.

**Comment by playpeace** [comment]
@jonte Pushed the changes as suggested.

File: docs/check_spelling.sh
Comment (tobsan) [inline docs/check_spelling.sh]: Sorry for being picky, but please add 2017 as year here.

**Review by jonte — APPROVED** [review]

**Pull request merged**
